tarxx.h

```diff
@@ -136,23 +221,38 @@ namespace tarxx {
 
         void write_header(const std::string& filename)
         {
+#ifdef __linux
+            struct ::stat buffer {
+            };
+            const auto stat_result = ::stat(filename.c_str(), &buffer);
+            if (stat_result != 0) {
+                throw errno_exception();
+            }
+
+            write_header(filename, buffer.st_mode & ALLPERMS, buffer.st_uid, buffer.st_gid, buffer.st_size, buffer.st_mtim.tv_sec);
+#endif
+        }
+
+#ifdef __linux
+        void write_header(const std::string& filename, __mode_t mode, __uid_t uid, __gid_t gid, __off_t size, __time_t time)
+        {
             if (type_ != tar_type::unix_v7) throw std::logic_error("unsupported tar format");
 
             block_t header {};
 
             write_into_block(header, filename, 0, 100);
 
-#ifdef __linux
-            struct ::stat buffer {};
+            struct ::stat buffer {
```

Comment:
@thilo-schmitt I'm not sure this change is intended. I used `clang-format-12` to format my PR. Is this correct?